apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/assessmentQuestion.ts

```diff
@@ -136,6 +140,32 @@ router.post(
       } else {
         res.send({});
       }
+    } else if (req.body.__action === 'bot_grade_assessment') {
+      // check if bot grading is enabled
+      const bot_grading_enabled = await features.enabledFromLocals('bot-grading', res.locals);
+      if (!bot_grading_enabled) {
+        throw new error.HttpStatusError(403, 'Access denied (feature not available)');
+      }
+
+      if (config.openAiApiKey === null || config.openAiOrganization === null) {
+        throw new error.HttpStatusError(501, 'Not implemented (feature not available)');
+      }
+
+      const jobSequenceId = await botGrade({
+        openaiconfig: {
+          apiKey: config.openAiApiKey,
+          organization: config.openAiOrganization,
+        },
```

Comment:
Would it make sense to read the config options in the lib instead of here?

Reply:
Our current thoughts is to throw an http error when apikey and organization are not given, which requires it to be outside the library function. Then since config is accessed in the page, we might just pass in the config to the library function. 
Maybe one thing we can do instead is to not use an http error, but throw an error in the server job?

Reply:
That is perfect!